tor_core/initialize.py

```diff
@@ -211,17 +213,67 @@ def initialize(config):
     logging.debug('Gifs loaded.')
 
 
-def build_bot(name, log_name='transcribersofreddit.log', require_redis=True):
+def get_heartbeat_port(config):
+    """
+    Attempts to pull an existing port number from the filesystem, and if it
+    doesn't find one then it generates the port number and saves it to a key
+    file.
+
+    :param config: the global config object
+    :return: int; the port number to use.
+    """
+    try:
+        # have we already reserved a port for this process?
+        port = int(open('heartbeat.port', 'r').readline().strip())
+        logging.debug('Found existing port saved on disk')
+        return port
+    except OSError:
+        pass
+
+    while True:
+        port = random.randrange(40000, 40200)  # is 200 ports too much?
```

Comment:
Not too many at all. I'm sure if someone is building over 200 bots on one server, there are already some issues with scaling that they would need to address first.